pkg/kcp/provider/aws/vpcpeering/new.go

```diff
@@ -26,6 +26,8 @@ func New(stateFactory StateFactory) composed.Action {
 				composed.ComposeActions("awsVpcPeering-non-delete",
 					addFinalizer,
 					loadVpc,
+					loadRemoteVpc,
+					createVpcPeeringConnection,
```

Comment:
Before a peering connection is made, it first should be checked if one already exists